src/coreComponents/linearAlgebra/interfaces/hypre/HypreVector.cpp

```diff
@@ -396,20 +233,20 @@ void HypreVector::pointwiseProduct( HypreVector const & x,
   GEOSX_LAI_ASSERT_EQ( localSize(), x.localSize() );
   GEOSX_LAI_ASSERT_EQ( localSize(), y.localSize() );
 
-  real64 const * const data = extractLocalVector();
-  real64 const * const x_data = x.extractLocalVector();
-  real64 * const y_data = y.extractLocalVector();
-  forAll< hypre::execPolicy >( localSize(), [y_data, data, x_data] GEOSX_HYPRE_DEVICE ( localIndex const i )
+  arrayView1d< real64 const > my_values = m_values.toViewConst();
```

Comment:
All of these can be `const`
```suggestion
  arrayView1d< real64 const > const my_values = m_values.toViewConst();
```